ticket.go

```diff
@@ -1,52 +1,73 @@
 package ticket
 
-// const (
-// 	Open int = iota
-// 	InProgress
-// 	Closed
-// )
-
-var StatusOpen = "Open"
-
-/*
-Why Doesnt the map work when used in NewTicket function:
-i.e.
-	Status:      StatusOpen[Open],
-
-var StatusOpen = map[int]string{
-	Open:       "open",
-	InProgress: "In Progress",
-	Closed:     "Complete",
-}
-*/
+import (
+	"fmt"
+)
+
+const (
+	StatusOpen int = iota
+	StatusClosed
+)
+
 type Ticket struct {
 	Subject     string
 	Description string
 	ID          int
-	Status      string
+	Status      int
+}
+
+type ticketMap map[int]*Ticket
+
+// Store stores tickets.
+type Store struct {
+	highestID int
+	tickets   ticketMap
 }
 
-type Project struct {
-	Name            string
-	highestID       int
-	ProjDescription string
+// NewStore returns a pointer to a new Store.
+func NewStore() *Store {
+	return &Store{
+		tickets: ticketMap{},
+	}
+}
+
+func (s *Store) Get(ID int) (*Ticket, error) {
+	tk, ok := s.tickets[ID]
+	if !ok {
+		return &Ticket{}, fmt.Errorf("no such ID %d", ID)
+	}
+	return tk, nil
 }
 
-func NewProject(name string) *Project {
-	return &Project{
-		Name: name,
+func (s *Store) NewTicket(subject string) *Ticket {
+	s.highestID++
+	tk := &Ticket{
+		Subject: subject,
+		ID:      s.highestID,
+		Status:  StatusOpen,
 	}
+	//save ticket here in a map:
+	s.tickets[tk.ID] = tk
+
+	//then return ticket:
+	return tk
 }
 
-func (p *Project) NewTicket(s string) Ticket {
-	p.highestID++
-	p.ProjDescription = "Pixels missing!"
+func (s *Store) GetByStatus(Status int) (tix []*Ticket, err error) {
+	result := []*Ticket{}
+	for _, ticket := range s.tickets {
+		if ticket.Status == StatusOpen {

```

Comment:
But this must be wrong, mustn't it? We should be comparing `ticket.Status` with the passed-in parameter `Status`. Instead, we only find tickets with `StatusOpen`, regardless of what status we were actually asked to find!

The fact that this happens to pass the test by accident is a useful signal to us that the test needs to be extended. What test would have caught this bug? Only one which called `GetByStatus()` with at least two _different_ status values. See the test itself for comments about this.